packages/legend-query/src/components/QueryBuilderValueSpecificationEditor.tsx

```diff
@@ -605,11 +1159,22 @@ export const DateInstanceValueEditor = observer(
 
 export const QueryBuilderValueSpecificationEditor: React.FC<{
   valueSpecification: ValueSpecification;
+  queryBuilderState: QueryBuilderState;
```

Comment:
I dont like `queryBuilderState` being added as a parameter here. 
`graph` should be enough to do everything you need for the editor. 
Stuff like `buildPrimitiveInstanceValue` can be refactored just to take a `PureModel`

Reply:
That's fair point, we could make this generic `ValueSpecEditor` if we can remove `queryBuilderState`